build-tools/tasks/a11y.js

```diff
@@ -4,8 +4,19 @@ const execa = require('execa');
 const glob = require('glob');
 const waitOn = require('wait-on');
 const { task } = require('../utils/gulp-utils.js');
+const { last } = require('lodash');
 
+function extractCurrentRunningShard() {
```

Comment:
This function assumes that the last parameter is always the shard, which will not always be the case. I'd recommend to extract the parameter by name instead, for example something like:
```js
const shardParameter = process.argv.find(arg => arg.startsWith("--shard="))
```

then we can also remove the `console.warn`.

Reply:
Can we pass the shard definition via env variables? 

Or if you want to have a CLI flag, here is the official recipe: https://github.com/gulpjs/gulp/blob/master/docs/recipes/pass-arguments-from-cli.md (but it is an extra dependency)

Reply:
the `parseArgs` function looks great but how do we feel if we used a polyfill until we update our workflows to use nodejs 18?

Reply:
It is linked on the node.js documentation page, so it is sort of an official recommendation

Reply:
The function seems to also be available in Node 16 (which we use in our dry-run):
https://nodejs.org/docs/latest-v16.x/api/util.html#utilparseargsconfig
So I think we won't need the polyfill.

Reply:
Nice!

Reply:
I used the `parseArgs` but I passed a `false` to the strict mode in its properties, reasons for that:

If the provided arguments in the `process.argv` does not have a matching key in the provided `options` it will throw an error, one parameter that gets passed is the `test:a11y`, I can also put in the options to overcome the error but using the `strict: false` is just more straightforward instead of placing an unused option.

Reply:
ok, seems appropriate for this use-case